src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs

```diff
@@ -85,6 +85,33 @@ public static string CreateEncodedSignature(string input, SigningCredentials sig
             }
         }
 
+        /// <summary>
+        /// Decompress JWT token bytes.
+        /// </summary>
+        /// <param name="tokenBytes"></param>
+        /// <param name="algorithm"></param>
+        /// <exception cref="ArgumentNullException">if 'tokenBytes' is null.</exception>
+        /// <exception cref="ArgumentNullException">if 'algorithm' is null.</exception>
+        /// <exception cref="NotSupportedException">if the decompression algorithm is not supported.</exception>
+        /// <returns>Decompressed JWT token</returns>
+        public static string DecompressToken(byte[] tokenBytes, string algorithm)
```

Comment:
If we make this public, then users have no easy way to add a new CompressionProviderFactory.

Reply:
Good idea. I can see how this doesn't give users any choice but to use the default compression provider. I've made this internal.